app/src/org/commcare/models/database/user/models/CommCareEntityStorageCache.java

```diff
@@ -163,4 +195,90 @@ public static int getEntityCacheWipedPref(String uuid) {
         return CommCareApplication.instance().getCurrentApp().getAppPreferences()
                 .getInt(uuid + "_" + ENTITY_CACHE_WIPED_PREF_SUFFIX, -1);
     }
+
+    public void primeCache(Hashtable<String, AsyncEntity> entitySet, String[][] cachePrimeKeys,
+            Detail detail) {
+        Vector<Integer> sortKeys = new Vector<>();
+        String validKeys = buildValidKeys(sortKeys, detail.getFields());
+        if ("".equals(validKeys)) {
+            return;
+        }
+
+        //Create our full args tree. We need the elements from the cache primer
+        //along with the specific keys we wanna pull out
+
+        String[] args = new String[cachePrimeKeys[1].length + sortKeys.size()];
+        System.arraycopy(cachePrimeKeys[1], 0, args, 0, cachePrimeKeys[1].length);
+
+        for (int i = 0; i < sortKeys.size(); ++i) {
+            args[cachePrimeKeys[1].length + i] = getCacheKey(detail.getId(), String.valueOf(sortKeys.get(i)));
+        }
+
+        String[] names = cachePrimeKeys[0];
+        String whereClause = buildKeyNameWhereClause(names);
+        long now = System.currentTimeMillis();
+        String sqlStatement = "SELECT entity_key, cache_key, value FROM entity_cache JOIN AndroidCase ON entity_cache.entity_key = AndroidCase.commcare_sql_id WHERE " +
+                whereClause + " AND " + CommCareEntityStorageCache.COL_APP_ID + " = '" + AppUtils.getCurrentAppId() +
+                "' AND cache_key IN " + validKeys;
+        SQLiteDatabase db = CommCareApplication.instance().getUserDbHandle();
+        if (SqlStorage.STORAGE_OUTPUT_DEBUG) {
+            DbUtil.explainSql(db, sqlStatement, args);
+        }
+
+        populateEntitySet(db, sqlStatement, args, entitySet);
+
+        if (SqlStorage.STORAGE_OUTPUT_DEBUG) {
+            Log.d(TAG, "Sequential Cache Load: " + (System.currentTimeMillis() - now) + "ms");
+        }
+    }
+
+    public String getCacheKey(String detailId, String mFieldId) {
+        return detailId + "_" + mFieldId;
+    }
+
+    private static String buildValidKeys(Vector<Integer> sortKeys, DetailField[] fields) {
```

Comment:
`buildValidKeys`, `buildKeyNameWhereClause` and `populateEntitySet` are private static methods called by non-static methods, do they need to be `static`?

Reply:
I prefer to make methods `private static` when they can be to signify that none of these methods changes the member variables of the class.